### PR TITLE
Beam archetypes: enable snapshot repositories.

### PR DESCRIPTION
**Pull request by peihe**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by peihe** [comment]
R: @davorbonaci

File: sdks/java/maven-archetypes/examples/src/main/resources/archetype-resources/pom.xml [hunk 1]
```diff
@@ -26,6 +26,20 @@
 
   <packaging>jar</packaging>
 
+  <repositories>
+    <repository>
+      <id>apache.snapshots</id>
+      <name>Apache Development Snapshot Repository</name>
+      <url>https://repository.apache.org/content/repositories/snapshots/</url>
+      <releases>
+        <enabled>true</enabled>
```
Comment (davorbonaci) [inline sdks/java/maven-archetypes/examples/src/main/resources/archetype-resources/pom.xml]: can/should this one be false?

**Comment by peihe** [comment]
PTAL

**Comment by davorbonaci** [comment]
LGTM. Will merge as soon as tests pass.

**Pull request closed without merging**